tests/test-device.toit

```diff
@@ -33,6 +52,38 @@ main args:
 
   cmd.run args
 
+start-backdoor --storage/Storage --device/service.Device:
+  network := net.open
+  // Listen on a free port.
+  tcp-socket := network.tcp-listen 0
+  print "$BACKDOOR-PREFIX http://localhost:$tcp-socket.local-address.port"
+  print BACKDOOR-FOOTER
+  server := http.Server
+  task::
+    server.listen tcp-socket:: | request/http.RequestIncoming writer/http.ResponseWriter |
+      resource := request.query.resource
+      if resource == "/device-id":
+        writer.out.write (json.encode "$device.id")
+      else if resource.starts-with "/storage/":
```

Comment:
Maybe pull this handling out in a helper method?

Reply:
Created a Backdoor class and moved the individual parts into a function.
done.